### DISCUSSION TITLE
Pug doesn't working.

### DISCUSSION THREAD

**Discussion by seventh-root** [quasarframework/quasar#14222] [Quasar App Extensions] [answered]
Hi there,

I've read many articles, but I couldn't find a solution, so let me ask you guys a question.
I have been creating my app with quasar.
And I was able to use pug with quasar version 2 in other App , but after upgrading to 3, pug is no longer available.

```quasar -v 3.5.7```

```
      chainWebpackMain(chain) {
        // do something with the Electron main process Webpack cfg
        // extendWebpackMain also available besides this chainWebpackMain
        chain.module.rules.push({
          test: /\.pug$/,
          loader: 'pug-plain-loader'
        })
      },
```

![image](https://user-images.githubusercontent.com/56397189/185668872-0b4fdd30-b65c-443f-a616-0a9b1c0e42ac.png)

**Comment by MilosPaunovic** [comment]
I don't know what you've installed, but latest version of Quasar is currently `2.7.7`, so the first step would be to install that one and try again.

**Reply by seventh-root** [reply]
Thank you Milos,
I have check version again.
So it was `2.7.5` and what I am using pug plugin are:
    "pug": "^3.0.2",
    "pug-plain-loader": "^1.1.0"

Do you have any information?

**Comment by metalsadman** [comment, marked as answer]
you should put the config inside `build.chainWebpack()` function.
```js
// quasar.conf.js
module.exports = function (ctx) {
  return {
    ...
    build: {
       ...
      chainWebpack(chain) {
        // here
```

**Reply by seventh-root** [reply]
@metalsadman 
Thank you so much.
It worked !
The problem is that I wrote code outside `build.chainWebpack()` you said.
So I fixed it.
![image](https://user-images.githubusercontent.com/56397189/185838835-3bec92e2-0f31-4e6d-b46c-c07ab5b26812.png)
